### PR TITLE
Dev ubec

### PR DESCRIPTION
**Pull request by tilschaef**
- biocontainers & tool upgrades
- New parameter syntax
- Refactored HTSeq count strandness
- small fixes

### CODE REVIEW

File: GATK/4.1.3.0/SplitNCigarReads.nf [hunk 1]
```diff
@@ -3,21 +3,18 @@ process SplitNCigarReads {
     tag {"GATK SplitNCigarReads ${sample_id}"}
     label 'GATK_4_1_3_0'
     label 'GATK_4_1_3_0_SplitNCigarReads'
-    clusterOptions = workflow.profile == "sge" ? "-l h_vmem=${params.splitncigarreads_mem}" : ""
```
Comment (sawibo) [inline GATK/4.1.3.0/SplitNCigarReads.nf]: change to params.mem?

File: GATK/4.1.3.0/SplitNCigarReads.nf [hunk 2]
```diff
@@ -3,21 +3,18 @@ process SplitNCigarReads {
...
-      file(genome_dict)
-
+   
     output:
       tuple sample_id, file("${sample_id}.split.bam"), file("${sample_id}.split.bai")
 
     script:
     """
-    gatk --java-options "-Xmx${task.memory.toGiga()-4}g -Djava.io.tmpdir=\$TMPDIR" \
+    gatk --java-options "-Xmx${task.memory.toGiga()-4}g -Djava.io.tmpdir=\$PWD" \
```
Comment (rernst) [inline GATK/4.1.3.0/SplitNCigarReads.nf]: Why? I think we should not use $PWD for tmp files.

File: GATK/4.1.3.0/SplitNCigarReads.nf (hunk 1 above)
Comment (tilschaef) [reply GATK/4.1.3.0/SplitNCigarReads.nf]: fixed

File: Sambamba/0.6.8/Index.nf [hunk 3]
```diff
@@ -2,7 +2,7 @@ process Index {
   tag {"SAMBAMBA_Index ${sample_id}"}
   label 'SAMBAMBA_0_6_8'
   label 'SAMBAMBA_0_6_8_Index'
-  clusterOptions = workflow.profile == "sge" ? "-l h_vmem=${params.index_mem}" : ""
+  clusterOptions = workflow.profile == "sge" ? "-l h_vmem=${params.sambambaindex.mem}" : ""
```
Comment (sawibo) [inline Sambamba/0.6.8/Index.nf]: change to params.mem

File: RSeQC/3.0.1/RSeQC.nf [hunk 4]
```diff
@@ -15,14 +15,13 @@ process RSeQC {
...
-    def inner_distance_arg = !params.singleEnd ? "inner_distance.py -i $bam -o ${bam.baseName}.rseqc -r $genome_bed12" :''
-    def read_distribution_arg = !params.singleEnd ? "read_distribution.py -i $bam -r $genome_bed12 > ${bam.baseName}.read_distribution.txt" :''
     """
+    inner_distance.py -i $bam -o ${bam.baseName}.rseqc -r $genome_bed12
+    read_distribution.py -i $bam -r $genome_bed12 > ${bam.baseName}.read_distribution.txt
     infer_experiment.py -i $bam -r $genome_bed12 > ${bam.baseName}.infer_experiment.txt
     junction_annotation.py -i $bam -o ${bam.baseName}.rseqc -r $genome_bed12
     bam_stat.py -i $bam 2> ${bam.baseName}.bam_stat.txt
     junction_saturation.py -i $bam -o ${bam.baseName}.rseqc -r $genome_bed12 2> ${bam.baseName}.junction_annotation_log.txt
     read_duplication.py -i $bam -o ${bam.baseName}.read_duplication
```
Comment (rernst) [inline RSeQC/3.0.1/RSeQC.nf]: `${bam}`

File: GATK/4.1.3.0/SplitNCigarReads.nf (hunk 2 above)
Comment (tilschaef) [reply GATK/4.1.3.0/SplitNCigarReads.nf]: Will try that one

Comment (sawibo) [reply GATK/4.1.3.0/SplitNCigarReads.nf]: I tend to agree with Robert but does this also work correctly on sge ? If not it would be a temporary fix

Comment (tilschaef) [reply GATK/4.1.3.0/SplitNCigarReads.nf]: For now, I reverted back to $TMPDIR but will do some testing and might come back if there are any issues.

File: Sambamba/0.6.8/Index.nf (hunk 3 above)
Comment (tilschaef) [reply Sambamba/0.6.8/Index.nf]: fix

File: RSeQC/3.0.1/RSeQC.nf (hunk 4 above)
Comment (tilschaef) [reply RSeQC/3.0.1/RSeQC.nf]: fixed

File: Sambamba/0.6.8/Index.nf (hunk 3 above)
Comment (sawibo) [reply Sambamba/0.6.8/Index.nf]: params.mem :) not params.sambambaindex.mem

Comment (tilschaef) [reply Sambamba/0.6.8/Index.nf]: ahhhhhh

**Pull request merged**